api/scenario_api_simulator/CMakeLists.txt

```diff
@@ -1,57 +1,51 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(scenario_api_simulator)
 
-add_compile_options(-std=c++14)
-
-find_package(catkin REQUIRED COMPONENTS
-  sensor_msgs
-  geometry_msgs
-  std_msgs
-  autoware_planning_msgs
-  autoware_system_msgs
-  autoware_perception_msgs
-  autoware_vehicle_msgs
-  npc_simulator
-  roscpp
-  tf2_ros
-  pcl_ros
-  pcl_conversions
-  lanelet2_extension
-  scenario_api_utils
-)
-
-catkin_package(
- INCLUDE_DIRS include
- LIBRARIES scenario_api_simulator
- CATKIN_DEPENDS scenario_api_utils sensor_msgs geometry_msgs std_msgs autoware_planning_msgs autoware_system_msgs autoware_perception_msgs autoware_vehicle_msgs npc_simulator roscpp tf2_ros pcl_ros pcl_conversions lanelet2_extension
-)
-
-include_directories(
-include
-${catkin_INCLUDE_DIRS}
-)
+### Compile options
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wno-unused-parameter -Wall -Wextra -Wpedantic)
+endif()
+
+find_package(ament_cmake REQUIRED)
+find_package(autoware_lanelet2_msgs REQUIRED)
+find_package(autoware_perception_msgs REQUIRED)
+find_package(dummy_perception_publisher REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(lanelet2_extension REQUIRED)
+find_package(npc_simulator REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(scenario_api_utils REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(unique_identifier_msgs REQUIRED)
```

Comment:
Are there any reason why you didn't use `ament_auto_find_build_dependencies()`?

Reply:
I get 
```
CMake Error at /opt/ros/foxy/share/ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake:66 (message):
  ament_target_dependencies() the passed package name 'npc_simulator' was not
  found before
```

Reply:
@nnmm I revamped the CMakeLists.txt and don't get that error but have `npc_simulator` installed so `COLCON_IGNORE` doesn't hurt. I think if you just wait for that to be merged (it's ready btw), then this will do

```cmake
cmake_minimum_required(VERSION 3.5)
project(scenario_api_simulator)

### Compile options
if(NOT CMAKE_CXX_STANDARD)
  set(CMAKE_CXX_STANDARD 14)
endif()
if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
  add_compile_options(-Wno-unused-parameter -Wall -Wextra -Wpedantic)
endif()

find_package(ament_cmake_auto REQUIRED)
ament_auto_find_build_dependencies()

ament_auto_add_library(scenario_api_simulator SHARED
  src/scenario_api_simulator.cpp
  src/npc_route_manager.cpp
)

ament_auto_package()
```
but it doesn't export targets, if you need that